### PR TITLE
xalan-c: Add --with-docs option

### PR DESCRIPTION
**Pull request by rleigh-dundee**
Not enabled by default due to the large size.

### CODE REVIEW

File: Library/Formula/xalan-c.rb [hunk 1]
```diff
@@ -30,6 +33,18 @@ def install
...
+        ENV.prepend_path "PATH", "#{buildpath}/c/bin"
+        cd "xdocs" do
+          # Set the library path in the script which runs Xalan from
+          # the source tree, or else the libxalan-c.dylib won't be found.
+          inreplace "sources/make-xalan.sh", "\"${XALANCMD}\" \\",
+                    "export DYLD_FALLBACK_LIBRARY_PATH=#{buildpath}/c/lib:$DYLD_FALLBACK_LIBRARY_PATH\n\"${XALANCMD}\" \\"
+          system "./make-apiDocs.sh"
+        end
+        mkdir_p "#{prefix}/share/doc"
+        system "mv", "build/docs/xalan-c", "#{prefix}/share/doc"
```
Comment (bfontaine) [inline Library/Formula/xalan-c.rb]: You can use `(prefix/"share/doc").install "build/docs/xalan-c"` instead of these two lines.

Comment (rleigh-dundee) [inline Library/Formula/xalan-c.rb]: Thanks, I've updated it do use this.  There's so much hidden functionality in homebrew I don't know about!

Comment (bfontaine) [inline Library/Formula/xalan-c.rb]: Yeah I was like you when I made my first contributions to this project :wink:

File: Library/Formula/xalan-c.rb [hunk 2]
```diff
@@ -11,8 +11,11 @@ class XalanC < Formula
     sha256 "fcfe6027b7d366f6a2bff783e0ab1e9abfc7c38c1a6fd31fa4a2fb9d325a2819" => :mavericks
   end
 
+  option "with-docs", "Install HTML docs"
   option :universal
 
+  depends_on "doxygen" => :build if build.with? "docs"
```
Comment (DomT4) [inline Library/Formula/xalan-c.rb]: You can block these:

```ruby
if build.with? "docs"
  depends_on "xyz"
  depends_on "abc"
end
```

Comment (rleigh-dundee) [inline Library/Formula/xalan-c.rb]: Thanks, I've done this.

**Comment by mikemcquaid** [comment]
> due to the large size.

How large?

**Comment by rleigh-dundee** [comment]
@mikemcquaid ~350MB vs 8MB.

File: Library/Formula/xalan-c.rb [hunk 3]
```diff
@@ -11,8 +11,13 @@ class XalanC < Formula
     sha256 "fcfe6027b7d366f6a2bff783e0ab1e9abfc7c38c1a6fd31fa4a2fb9d325a2819" => :mavericks
   end
 
+  option "with-docs", "Install HTML docs"
   option :universal
 
+  if build.with? "docs"
+    depends_on "doxygen" => :build if build.with? "docs"
```
Comment (mikemcquaid) [inline Library/Formula/xalan-c.rb]: Whoops, the `if build.with? "docs"` has been repeated twice.

File: Library/Formula/xalan-c.rb [hunk 4]
```diff
@@ -30,6 +35,17 @@ def install
                             "--disable-silent-rules",
                             "--prefix=#{prefix}"
       system "make", "install"
+      if build.with? "docs"
+        ENV.prepend_path "PATH", "#{buildpath}/c/bin"
+        cd "xdocs" do
+          # Set the library path in the script which runs Xalan from
+          # the source tree, or else the libxalan-c.dylib won't be found.
+          inreplace "sources/make-xalan.sh", "\"${XALANCMD}\" \\",
+                    "export DYLD_FALLBACK_LIBRARY_PATH=#{buildpath}/c/lib:$DYLD_FALLBACK_LIBRARY_PATH\n\"${XALANCMD}\" \\"
```
Comment (mikemcquaid) [inline Library/Formula/xalan-c.rb]: Can you please submit this patch to the upstream developers of this project and add a link to the upstream patch submission and explanation of why the patch is needed in a comment in the formula file. Thanks!

**Comment by DomT4** [comment]
Closing as dead. Please feel free to resubmit a fresh PR as desired. Thanks for your submission to Homebrew!

**Comment by rleigh-dundee** [comment]
@DomT4 Sorry, why was this considered dead?  I thought it was done and ready to merge weeks ago.  Was there something outstanding I was supposed to change?

**Comment by DomT4** [comment]
Mike's comments from further up were missed:

> Whoops, the `if build.with? "docs"` has been repeated twice.
> Can you please submit this patch to the upstream developers of this project and add a link to the upstream patch submission and explanation of why the patch is needed in a comment in the formula file. Thanks!

Happy to reopen if you want to address those points?

**Comment by rleigh-dundee** [comment]
@DomT4 Ah, I totally missed that.  If you'd like to reopen it I'll fix that up shortly.  Thanks!

File: Library/Formula/xalan-c.rb (hunk 3 above)
Comment (rleigh-dundee) [inline Library/Formula/xalan-c.rb]: This is now fixed.  Sorry for the delay; I must have missed the notification of your comment.

**Comment by bfontaine** [comment]
@rleigh-dundee Did you report the `inreplace` thing to upstream?

**Comment by rleigh-dundee** [comment]
@bfontaine I can't remember offhand.  I'll check and do so if not.

**Comment by bfontaine** [comment]
Thanks!

**Comment by rleigh-dundee** [comment]
@bfontaine https://issues.apache.org/jira/browse/XALANC-766

**Comment by bfontaine** [comment]
@rleigh-dundee Awesome, thank you!

**Comment by rleigh-dundee** [comment]
All green now.

**Comment by mikemcquaid** [comment]
Thanks @rleigh-dundee!

**Pull request closed without merging**